### PR TITLE
Dependabot added

### LINKED ISSUE #231: [Feature Request]: Add dependabot for keeping dependencies up to date
### How would you like to enhance the existing project.

This would keep the dependencies up to date and ensure there are no dependency conflicts

### Describe breifly how would you implement the same.

This would require creating a new GitHub action which will check the dependency version and update them 

### Are you contributing for a program?

GSSoC23

### Code of Conduct

- [X] I follow [CONTRIBUTING GUIDELINE](https://github.com/amupedia2021/Project-Amupedia/blob/master/CONTRIBUTING.md) of this project.

### PR DESCRIPTION
**Pull request by Tushar98644**
## Description
fixes #231 

@AmanJain18 @tusharbansal22 I have added the dependabot which will keep track of all the dependencies

## Pull Request Checklist

Please make sure that your PR meets the following requirements:

- [x] The app is working correctly after applying the changes.
- [x] The app is responsive and displays properly on different screen sizes and devices.
- [x] The changes are thoroughly documented in the appropriate files or documentation sections.
- [x] The coding conventions and style guidelines are followed consistently.

## Checklist

Please check all applicable boxes and provide any additional information if needed.

- [x] I have tested the changes locally and ensured that the app is functioning as expected.
- [x] I have verified that the app is responsive and looks good on various screen sizes and devices.
- [x] I have reviewed the documentation and ensured that it accurately reflects the changes made.
- [x] I have followed the coding conventions and style guidelines defined in the project.
- [x] I have run the automated tests and all of them pass.
- [x] I have considered the performance implications of the changes made.
- [x] I have considered the security implications of the changes made.
- [x] I have checked for any potential regressions or side effects caused by the changes.
- [x] I have performed a self-review of my code.
- [x] Starred the repository.
- [x] Added My Name to the Contributor's List.
- [x] Followed the repository's [Contributing Guidelines](/CONTRIBUTING.md).
- [x] I ran the app and tested it locally to verify that it works as expected.

### CODE REVIEW

**Review by tusharbansal22 — APPROVED** [review]
Looks ok to. me. Can merge  if @anmode @AmanJain18 approve

**Comment by anmode** [comment]
cool! Can we add other bots in this PR too? like as greeting bot?

File: .github/dependabot.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+version: 2
+updates:
+
+  - package-ecosystem: "npm"
+    directory: "/"
+    schedule:
+      interval: "daily"
```
Comment (anmode) [inline .github/dependabot.yaml:7]: change it to weekly...or 14 days would be good option

**Review by anmode — CHANGES_REQUESTED** [review]
cool

**Comment by Tushar98644** [comment]
> cool! Can we add other bots in this PR too? like as greeting bot?

should I create another issue for that? this would help keep the pr's clean and specific

File: .github/dependabot.yaml (hunk 1 above)
Comment (Tushar98644) [reply .github/dependabot.yaml]: Dependabot currently supports the following interval values: "daily", "weekly", and "monthly". There is no specific interval value for every 14 days. I have changed it to weekly

**Review by anmode — APPROVED** [review]

**Comment by anmode** [comment]
> > cool! Can we add other bots in this PR too? like as greeting bot?
> 
> should I create another issue for that? this would help keep the pr's clean and specific

I was thinking in this..ok create one bot pR. we will continue all other bots in that

**Comment by AmanJain18** [comment]
> > > cool! Can we add other bots in this PR too? like as greeting bot?
> > 
> > should I create another issue for that? this would help keep the pr's clean and specific
> 
> I was thinking in this..ok create one bot pR. we will continue all other bots in that

No need to create PR for bot RN, Will update it later...

**Pull request merged**